src/droid/droid-source.c

```diff
@@ -274,6 +274,55 @@ static void unsuspend(struct userdata *u) {
         u->stream_valid = false;
 }
 
+/* Called from IO context */
+static int source_set_state_in_io_thread_cb(pa_source *s, pa_source_state_t new_state, pa_suspend_cause_t new_suspend_cause) {
+    struct userdata *u;
+
+    pa_assert(s);
+    pa_assert_se(u = s->userdata);
+
+    /* It may be that only the suspend cause is changing, in which case there's
+     * nothing more to do. */
+    if (new_state == s->thread_info.state)
+        return 0;
+
+    switch (new_state) {
+        case PA_SOURCE_SUSPENDED: {
+            int r;
+
+            if (PA_SOURCE_IS_OPENED(u->source->thread_info.state)) {
+                if ((r = suspend(u)) < 0)
+                    return r;
+            }
+
+            break;
+        }
```

Comment:
indentation due to the otherwise necessary "extra" {} block looks  a bit confusing...